integration_tests/src/main/python/sort_test.py

```diff
@@ -40,8 +40,7 @@ def test_single_orderby(data_gen, order):
 @pytest.mark.parametrize('stable_sort', ['STABLE', 'OUTOFCORE'])
 @pytest.mark.parametrize('data_gen', [
     pytest.param(all_basic_struct_gen),
-    pytest.param(StructGen([['child0', all_basic_struct_gen]]),
```

Comment:
In the original code we didn't add any tests that would check that we fall back to the CPU on structs that have a mismatch between the null ordering, it would be nice if you could add it here.

Reply:
I've added a case to test fallback due to non-default null ordering.